IdentityCore/src/requests/sdk/adal/MSIDLegacySilentTokenRequest.m

```diff
@@ -100,6 +100,14 @@ - (BOOL)shouldRemoveRefreshToken:(NSError *)serverError
     return oauthError == MSIDErrorServerInvalidGrant;
 }
 
+- (BOOL)shouldRemoveAccountArtifacts:(NSError *)serverError
+{
+    // Removing account artifacts on invalid_grant + user_deleted_account suberror combination
+    MSIDErrorCode oauthError = MSIDErrorCodeForOAuthError(serverError.msidOauthError, MSIDErrorInternal);
```

Comment:
I see the implementation for this method is same in both subclasses, would it make sense to move it to super class ?

Reply:
I was thinking about this when I was implementing this and and thank you for pointing that up. Left it just in case there was a need but I guess after feature completion it makes sense to move to base class since those methods match exactly

Reply:
Hi @Veena11 another update while implementing this:
Seems that `[super shouldRemoveAccountArtifacts:serverError];` wouldn't compile as that function is not defined in the header: MSIDSilentTokenRequest.h. This would be like a private method to be exposed in public header if I add it for subclasses to see. The fix is to have an extra header+internal. Thinking not to make more changes there for this one function call and keep the change as is.

Reply:
I feel it is better to have the internal header file to avoid having duplicate code in multiple files ? Would the change be significant , may be I am missing something ?